Bottom: e4470945b5b3e073d6bd26f6a6cb3bfd4f874256
Top:    cd88d9fdde844f5987cce753bd199b3bff89367c
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2009-11-11 16:43:16 +0000

Temp fixes and diagnostics


---

diff --git a/src/polygon1.c b/src/polygon1.c
index c2526ae..6dc8b76 100644
--- a/src/polygon1.c
+++ b/src/polygon1.c
@@ -1145,7 +1145,11 @@ label_contour (PLINE * a)
           continue;
         }
     }
-  while ((cur = cur->next) != first_not_unknown);
+  while ((cur = cur->next) != first_not_unknown &&
+         !(cur->next == &a->head && first_not_unknown == NULL));
+
+  if (first_not_unknown == NULL)
+    fprintf (stderr, "OOPS, DIDN'T LABEL ANYTHING!\n");
 #ifdef DEBUG_ALL_LABELS
   print_labels (a);
   DEBUGP ("\n\n");
diff --git a/src/sweep.h b/src/sweep.h
new file mode 100644
index 0000000..e69de29
